lib/Sema/TypeCheckDecl.cpp

```diff
@@ -868,6 +868,12 @@ static void checkRedeclaration(TypeChecker &tc, ValueDecl *current) {
   if (!currentFile || currentDC->isLocalContext())
     return;
 
+  // Cannot define a type called 'Any'
+  if (current->getFullName() == tc.Context.Id_Any) {
```

Comment:
Thinking about it more I could remove the restriction of not being able to declare an Any type whether or not I change the parser — both solutions would require the user to reference their own types as `Module.Any` or `Outer.Any`.